### ISSUE TITLE
Aux window terminals have messed up cursor blinking

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#199466]
Testing #199023

May be an issue in terminal code, or related to tricks we try to do around `setTimeout`/`setInterval`?

![Recording 2023-11-28 at 15 33 15](https://github.com/microsoft/vscode/assets/2193314/36583330-4784-4537-9816-409f1e31bc6c)

**Comment by bpasero** [comment]
@Tyriar we now schedule `setTimeout` on every window that is opened but we cancel all other timeouts immediately once the first one runs (we cancel even before triggering the callback). I would be surprised if timers would run more than once. We even have an extra utility method to really ensure the callback is only run once:

https://github.com/microsoft/vscode/blob/2e721b6ec731c22cd3c2853dbc7ad69deb447515/src/vs/workbench/browser/window.ts#L68-L71

**Comment by Tyriar** [comment]
Thanks for the pointer, will investigate

**Comment by bpasero** [comment]
@Tyriar one thing that I noticed is a lot of `getActiveWindow` usages in terminal land. As part of https://github.com/microsoft/vscode/pull/199568 I have switched some terminal code to use `mainWindow` specifically for `setTimeout` when also `clearTimeout` is used. I think routing to the active window in these cases is dangerous because a timer may be installed on one window and then attempted to be cleared on another window, which will not work. Going via `mainWindow` will ensure everything is properly handled and dispatched.

However, I still found some disposable intervals being scheduled on the active window and wonder if we can rather try to pick the right window for where the terminal is in.

**tl;dr;**: avoid `getActiveWindow` or `getActiveDocument` where you can infer the right context based on DOM nodes you are building up.

**Comment by Tyriar** [comment]
Can no longer reproduce

**Comment by Tyriar** [comment]
Just saw it happen, not sure on repro though.
